kitsune/sumo/utils.py

```diff
@@ -315,6 +319,42 @@ def get_browser(user_agent):
     return browser
 
 
+def has_blocked_link(data):
+    # TODO: deal with punycode when presented in non-ascii form
+    for match in POTENTIAL_LINK_REGEX.finditer(data):
+        tld = match.group(1).upper()
+        if tld in VALID_TLDS:
+            full_domain = match.group(0).lower()
+            in_allowlist = False
+            for allowed_domain in settings.ALLOW_LINKS_FROM:
+                split = full_domain.rsplit(allowed_domain, 1)
+                if len(split) != 2:
+                    # allowed_domain isn't in full_domain, or something went wrong
+                    continue
+                if split[-1] != '':
```

Comment:
you can merge the conditionals which have the `continue` statement together with the `or` statement